From b04c0a86b19856071c29d2a6285f3240c606ee7a Mon Sep 17 00:00:00 2001
From: "H.J. Lu" <hjl.tools@gmail.com>
Date: Tue, 27 Apr 2021 09:08:41 -0700
Subject: [PATCH] Always include <config.h> in cipher assembly codes

* cipher/poly1305-s390x.S: Always include <config.h>.

When Intel CET is enabled, we need to include <cet.h> in assembly codes
to mark Intel CET support even if it is empty.  We should always include
<config.h> in cipher assembly codes so that they will be marked for
Intel CET support when compiling for x86-64 and i686.

Signed-off-by: H.J. Lu <hjl.tools@gmail.com>
---
 cipher/poly1305-s390x.S | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cipher/poly1305-s390x.S b/cipher/poly1305-s390x.S
index 844245f6..28bed560 100644
--- a/cipher/poly1305-s390x.S
+++ b/cipher/poly1305-s390x.S
@@ -18,8 +18,8 @@
  * License along with this program; if not, see <http://www.gnu.org/licenses/>.
  */
 
-#if defined (__s390x__) && __GNUC__ >= 4 && __ARCH__ >= 9
 #include <config.h>
+#if defined (__s390x__) && __GNUC__ >= 4 && __ARCH__ >= 9
 #if defined(HAVE_GCC_INLINE_ASM_S390X)
 
 #include "asm-poly1305-s390x.h"
-- 
GitLab

